ext/drutil/drutil.c

```diff
@@ -71,23 +71,31 @@ static int drutil_init_count;
 #ifdef X86
 
 static inline void
-native_unix_cpuid(uint *eax, uint *ebx, uint *ecx, uint *edx)
+native_unix_cpuid(uint *xax, uint *xbx, uint *xcx, uint *xdx)
 {
 #    ifdef UNIX
-    /* We need to do this ebx trick, because ebx might be used for fPIC,
+    /* We need to do this xbx trick, because xbx might be used for fPIC,
      * and gcc < 5 chokes on it. This can get removed and replaced by
      * a "=b" constraint when moving to gcc-5.
      */
-    asm volatile("xchgl\t%%ebx, %k1\n\t"
+#        ifdef X64
+    asm volatile("xchgq\t%%rbx, %q1\n\t"
```

Comment:
OK so the bug was that rbx's top 32 bits were cleared.  Please add a comment to the issue explaining that.

Reply:
Ok. Done.